frontend/src/pages/AccountSent.vue

```diff
@@ -12,61 +12,88 @@
       </div>
     </div>
     <div v-else class="q-mx-auto" style="max-width: 800px">
-      <account-sent-table :sendMetadata="sendMetadata" />
+      <div v-if="needsSignature" class="text-center q-mb-md">
+        {{ $t('AccountSent.need-signature') }}
+        <base-button @click="getData" class="text-center q-mt-md" :label="$t('AccountSent.sign')" />
+      </div>
+      <div className="flex" v-else-if="dataLoading">
+        <loading-spinner />
+      </div>
+      <div v-else-if="!needsSignature && !dataLoading" class="q-mx-auto" style="max-width: 800px">
+        <account-sent-table :sendMetadata="sendMetadata" />
+      </div>
     </div>
   </q-page>
 </template>
 
 <script lang="ts">
-import { defineComponent, ref } from 'vue';
+import { defineComponent, ref, computed, onMounted } from 'vue';
 import useWalletStore from 'src/store/wallet';
 import AccountSentTable from 'components/AccountSentTable.vue';
 import ConnectWallet from 'components/ConnectWallet.vue';
 import { SendTableMetadataRow } from 'components/models';
-import { BigNumber } from 'src/utils/ethers';
+import { BigNumber, getAddress, isHexString } from 'src/utils/ethers';
 import { formatNameOrAddress } from 'src/utils/address';
 import { formatDate, formatAmount, formatTime, getTokenSymbol, getTokenLogoUri } from 'src/utils/utils';
-import { NATIVE_TOKEN_ADDRESS } from 'components/models';
+import { fetchAccountSends } from 'src/utils/account-send';
 
 function useAccountSent() {
-  const { tokens, userAddress } = useWalletStore();
+  const { tokens, userAddress, chainId, viewingKeyPair, getPrivateKeys, provider } = useWalletStore();
   const sendMetadata = ref<SendTableMetadataRow[]>([]);
+  const dataLoading = ref<boolean>(false);
+  const needsSignature = computed(() => !viewingKeyPair.value?.privateKeyHex);
+  const viewingPrivateKey = computed(() => viewingKeyPair.value?.privateKeyHex);
 
-  const data = [
-    {
-      amount: '1000000000000000000',
-      tokenAddress: NATIVE_TOKEN_ADDRESS,
-      dateSent: new Date(),
-      address: '0x2436012a54c81f2F03e6E3D83090f3F5967bF1B5',
-      hash: '0xabeea5720640859db6dc46caae44fe34449469bfae8e3e930d9f59abbf50ed50',
-    },
-    {
-      amount: '2000000000000000000',
-      tokenAddress: NATIVE_TOKEN_ADDRESS,
-      dateSent: new Date('2021-01-01'),
-      address: '0xEAC5F0d4A9a45E1f9FdD0e7e2882e9f60E301156',
-      hash: '0xf1a4ad61bd073c8262f18a13b81f9a3ce33884d22963be3f9150719116ac3194',
-    },
-  ];
+  const getData = async () => {
+    if (needsSignature.value || !viewingPrivateKey.value) {
+      const success = await getPrivateKeys();
+      if (success === 'denied') return; // if unsuccessful, user denied signature or an error was thrown
+    }
+    if (!viewingKeyPair.value?.privateKeyHex) {
+      return;
```

Comment:
Similar question, if we're here doesn't this now have to be defined? I'm ok with having these redundancies as safety checks, but if that's the purpose let's add comments explaining it

Reply:
The reason we have this check is because I am able to get to the send page and send without having the viewing key defined. I am able to reproduce pretty consistently if it isn't expected behavior and we want to dive deeper.